spec/services/exports/export_adjustments_csv_service_spec.rb

```diff
@@ -0,0 +1,180 @@
+RSpec.describe Exports::ExportAdjustmentsCSVService, :wip do
+  # Create organization after items to ensure proper associations
+  let!(:item1) { create(:item, name: "item1") }
+  let!(:item2) { create(:item, name: "item2") }
+  let!(:item3) { create(:item, name: "item3") }
+  let!(:item4) { create(:item, name: "item4") }
+  let!(:item5) { create(:item, :inactive, name: "item5") }
+
+  # Now create organization and associate items with it
+  let(:organization) do
+    org = create(:organization)
+    [item1, item2, item3, item4, item5].each do |item|
+      item.update!(organization_id: org.id)
+    end
+    org
+  end
+
+  let(:sorted_item_names) do
+    [item1, item2, item3, item4, item5].map(&:name).sort
+  end
+
+  let(:expected_headers) do
+    [
+      "Created date", "Storage Area",
+      "Comment", "# of changes"
+    ] + sorted_item_names
+  end
+
+  let(:storage_location) { create(:storage_location, organization: organization) }
+  let(:user) { create(:user, organization: organization) }
+
+  describe "#generate_csv_data" do
+    subject { described_class.new(adjustments: adjustments, organization: organization).generate_csv_data }
+
+    context "with multiple adjustments and items" do
+      let(:adjustments) do
+        [
+          # 1st adjustment with 2 items
+          create(:adjustment,
+            user_id: user.id,
+            storage_location: storage_location,
+            organization: organization,
+            comment: "adjustment 1",
+            line_items_attributes: [
+              {item_id: item1.id, quantity: 10},
+              {item_id: item2.id, quantity: -5}
+            ]),
+
+          # 2nd adjustment with 1 item
+          create(:adjustment,
+            user_id: user.id,
+            storage_location: storage_location,
+            organization: organization,
+            comment: "adjustment 2",
+            line_items_attributes: [
+              {item_id: item3.id, quantity: 3}
+            ]),
+
+          # 3rd adjustment with the :with_items trait
+          create(:adjustment, :with_items,
+            user_id: user.id,
+            storage_location: storage_location,
+            organization: organization,
+            comment: "adjustment 3",
+            item: item1,
+            item_quantity: 7)
+        ]
+      end
+
+      it "should have the expected headers" do
+        expect(subject[0]).to eq(expected_headers)
+      end
+
+      it "should include the adjustment data in the rows" do
+        # Check 1st adjustment
+        expect(subject[1]).to include(
```

Comment:
Is there a reason we're using `include` instead of `eq`?

My comment above re CSV output applies here as well.

Reply:
1. ```eq``` would be what I meant for the assertions, will change, thanks :)

2. Will also change to more static csv-to-csv assertions here as well, thanks for the pointers